fix(oxlint/oxfmt/lsp): tell client the real tool name & version

Sysix opened the pull request:
for `oxfmt --lsp`:

```
2025-11-27 19:21:51.286 [info] [Trace - 7:21:51 PM] Received response 'initialize - (0)' in 276ms.
2025-11-27 19:21:51.286 [info] Result: {
    "capabilities": {
        "documentFormattingProvider": true,
        "textDocumentSync": {
            "change": 1,
            "openClose": true,
            "save": {
                "includeText": false
            }
        },
        "workspace": {
            "workspaceFolders": {
                "changeNotifications": true,
                "supported": true
            }
        }
    },
    "serverInfo": {
        "name": "oxfmt",
        "version": "0.15.0"
    }
}
```

camc314 commented:
### Merge activity

* **Nov 29, 4:56 PM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Nov 29, 4:56 PM UTC**: `camc314` added this pull request to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc).

Sysix committed:
fix(oxlint/oxfmt/lsp): tell client the real tool name & version (#16212)

for `oxfmt --lsp`:

```
2025-11-27 19:21:51.286 [info] [Trace - 7:21:51 PM] Received response 'initialize - (0)' in 276ms.
2025-11-27 19:21:51.286 [info] Result: {
    "capabilities": {
        "documentFormattingProvider": true,
        "textDocumentSync": {
            "change": 1,
            "openClose": true,
            "save": {
                "includeText": false
            }
        },
        "workspace": {
            "workspaceFolders": {
                "changeNotifications": true,
                "supported": true
            }
        }
    },
    "serverInfo": {
        "name": "oxfmt",
        "version": "0.15.0"
    }
}
```
